### PR TITLE
Fix doc strings for ASR related files

### PR DESCRIPTION
**Pull request by vectominist**
Fix doc strings for ASR related files, including:
- moving doc strings out of `__init__`
- adding `__all__` to some files
- change `G2P.__call__` to `G2P.encode`

### CODE REVIEW

File: s3prl/dataio/corpus/librispeech.py [hunk 1]
```diff
@@ -86,6 +86,17 @@ def _parse_spk_to_gender(speaker_file: Path) -> dict:
 
 
 class LibriSpeech(Corpus):
```
Comment (leo19941227) [inline s3prl/dataio/corpus/librispeech.py:88]: This change looks good to me, while I find that the `all_data` property not yet has doc. Could you follow the doc template in `fluent_speech_command` for librispeech? Thanks!

**Review by leo19941227 — CHANGES_REQUESTED** [review]
Cool @vectominist ! I only have one comment. Feel free to merge it after adding that part.

File: s3prl/dataio/corpus/librispeech.py (hunk 1 above)
Comment (vectominist) [reply s3prl/dataio/corpus/librispeech.py:88]: Thanks Leo! I fixed it, please take a look

**Review by leo19941227 — APPROVED** [review]
Looks good to me. Thanks! I believe you can merge this PR right? Please do that after the CI is passed.

**Pull request merged**